src/datachain/lib/file.py

```diff
@@ -206,18 +133,13 @@ def parent(self):
     @contextmanager
     def open(self, mode: Literal["rb", "r"] = "rb") -> Iterator[Any]:
         """Open the file and return a file object."""
-        if self.location:
```

Comment:
does it mean we can drop location field after this PR?

Reply:
Yup, it looks like we could. I opened to make #487 for that change, although since it's more "breaking," I'm not sure if/when it makes sense to merge.